src/VecSim/spaces/IP/IP_SVE_FP64.h

```diff
@@ -0,0 +1,72 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+
+#include <arm_sve.h>
+
+static void InnerProductStep(double *&pVect1, double *&pVect2, size_t &offset, svfloat64_t &sum) {
+    // Load vectors
+    svfloat64_t v1 = svld1_f64(svptrue_b64(), pVect1 + offset);
+    svfloat64_t v2 = svld1_f64(svptrue_b64(), pVect2 + offset);
+
+    // Multiply-accumulate
+    sum = svmla_f64_z(svptrue_b64(), sum, v1, v2);
+
+    // Advance pointers
+    offset += svcntd();
+}
+
+template <bool partial_chunk, unsigned char additional_steps>
+double FP64_InnerProductSIMD_SVE(const void *pVect1v, const void *pVect2v, size_t dimension) {
+    double *pVect1 = (double *)pVect1v;
+    double *pVect2 = (double *)pVect2v;
+    size_t offset = 0;
+
+    uint64_t vl = svcntd();
+
+    // Multiple accumulators to increase instruction-level parallelism
+    svfloat64_t sum0 = svdup_f64(0.0f);
```

Comment:
`0.0f` is explicit float32. Let's use doubles in all these implementations (`0.0` for this example)